### PR TITLE
[plugin.video.pt] v0.1.2

### PR DESCRIPTION
**Pull request by Haui1112**
### Description

This plugin makes it possible to browse peertube instances from https://instances.joinpeertube.com, list local videos per instance and play the videos on next gen kodi (post python2)

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.pt/main.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+ADDON_PATH = translatePath(Addon().getAddonInfo('path'))
+ICONS_DIR = os.path.join(ADDON_PATH, 'resources', 'images', 'icons')
+FANART_DIR = os.path.join(ADDON_PATH, 'resources', 'images', 'fanart')
+
+def get_url(**kwargs):
+    return '{}?{}'.format(URL, urlencode(kwargs))
+
+def get_instances():
+    filename = "instances.json"
+    if not os.path.isfile(filename):
```
Comment (basrieter) [inline plugin.video.pt/main.py]: You should only store files in the user data folder. You should use `xbmcvfs`  (See https://romanvm.github.io/Kodistubs/_autosummary/xbmcvfs.html) and use `xbmcvfs`:

```python
import os
import xbmcvfs

addon_id = 'plugin.video.example'
userdata_path = f'special://userdata/addon_data/{addon_id}/'
filename = 'example.txt'
file_path = os.path.join(userdata_path, filename)

with xbmcvfs.File(file_path) as f:
    f.write("content")
```

File: plugin.video.pt/addon.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  <import addon="script.module.requests" version="2.25.1"/>
+</requires>
+<extension point="xbmc.python.pluginsource" library="main.py">
+  <provides>video</provides>
+</extension>
+<extension point="xbmc.addon.metadata">
+  <summary lang="en_GB">Next Gen Peertube Addon</summary>
+  <description lang="en_GB">In lieu of a working peertube addon, I've decided to make a new one.</description>
+  <disclaimer lang="en_GB">Feel free to build and improve on this addon.</disclaimer>
+  <license>GPL-3.0-only	</license>
```
Comment (basrieter) [inline plugin.video.pt/addon.xml]: Please remove the extra `space` here.

File: plugin.video.pt/resources/images/icon.png
Comment (basrieter) [inline plugin.video.pt/resources/images/icon.png:1]: Please make the add-on checker happy with an icon without transparency.

**Review by basrieter — CHANGES_REQUESTED** [review]
Please see my comments.

**Comment by Haui1112** [comment]
Virtual filesystem and icon without transparency should be added now.

File: plugin.video.pt/addon.xml (hunk 2 above)
Comment (Haui1112) [reply plugin.video.pt/addon.xml]: Has been removed

File: plugin.video.pt/resources/images/icon.png
Comment (Haui1112) [reply plugin.video.pt/resources/images/icon.png:1]: New Icon has been added

File: plugin.video.pt/main.py (hunk 1 above)
Comment (Haui1112) [reply plugin.video.pt/main.py]: virtual filesystem is now being used

File: plugin.video.pt/main.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+def get_instances():
+    filename = "instances.json"
+    if not os.path.exists(USERDATA_PATH):
+        os.makedirs(USERDATA_PATH)        
+    FILE_PATH = os.path.join(USERDATA_PATH, filename)
+    if not xbmcvfs.exists(FILE_PATH):
+        print("No file, requesting new data!")
+        request = requests.get('https://instances.joinpeertube.org/api/v1/instances/hosts?count=1000&start=0&sort=createdAt')
+        r = request.json()
+        with open(FILE_PATH, 'w', encoding='utf-8') as instances_file:
```
Comment (basrieter) [inline plugin.video.pt/main.py]:
```python
with xbmcvfs.File(file_path) as instances_file:
    json.dump(r, instances_file, ensure_ascii=False, indent=4)
```

and if `json.dump()` does not like it:

```python
with xbmcvfs.File(file_path) as instances_file:
    instances_file.write(json.dumps(r, ensure_ascii=False, indent=4)
```

File: plugin.video.pt/main.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+            json.dump(r, instances_file, ensure_ascii=False, indent=4)
+
+    else:
+        r = json.load(open(FILE_PATH))
```
Comment (basrieter) [inline plugin.video.pt/main.py]: Als here:

```python
with xbmcvfs.File(file_path) as instances_file:
    json.load(instance_file)
```

the `with` is essential to close all handles.

**Review by basrieter — CHANGES_REQUESTED** [review]
Still a bit of file handling issues.

File: plugin.video.pt/main.py (hunk 3 above)
Comment (Haui1112) [reply plugin.video.pt/main.py]: added as requested.

File: plugin.video.pt/main.py (hunk 4 above)
Comment (Haui1112) [reply plugin.video.pt/main.py]: added as requested

**Review by basrieter — APPROVED** [review]
Thanks for your effort.

**Pull request merged**